staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -31,11 +31,15 @@ import (
 	"k8s.io/client-go/rest"
 )
 
-type dynamicClient struct {
-	client *rest.RESTClient
+type DynamicClient struct {
+	client rest.Interface
 }
 
-var _ Interface = &dynamicClient{}
+func (c *DynamicClient) RESTClient() rest.Interface {
```

Comment:
strong -1 on exposing this... this was a mistake for the other clients that breaks encapsulation via an interface and makes them ~unmockable (xref https://github.com/kubernetes/kubernetes/issues/26130)

Reply:
Yikes, that is not ideal. I had no idea - I'll back this out, but since the rest of the clients are locked in to that behavior, consistency would at least be nice.